src/dotnet-compile-fsc/Program.cs

```diff
@@ -295,5 +278,100 @@ private static IEnumerable<string> GetSourceFiles(IReadOnlyList<string> sourceFi
 
             yield return sourceFiles.Last();
         }
+
+        private static Command RunFsc(List<string> fscArgs, string temp)
+        {
+            var fscEnvExe = Environment.GetEnvironmentVariable("DOTNET_FSC_PATH");
+            var exec = Environment.GetEnvironmentVariable("DOTNET_FSC_EXEC")?.ToUpper() ?? "COREHOST";
+            
+            var muxer = new Muxer();
+
+            if (fscEnvExe != null)
+            {
+                switch (exec)
+                {
+                    case "RUN":
+                        return Command.Create(fscEnvExe, fscArgs.ToArray());
+
+                    case "COREHOST":
+                    default:
+                        var host = muxer.MuxerPath;
+                        return Command.Create(host, new[] { fscEnvExe }.Concat(fscArgs).ToArray());
```

Comment:
Existing code here by enrico, trying to have as little impact on dotnet-compile-fsc as possible because we will be taking this out of the repo.